sui_programmability/examples/basics/sources/Sandwich.move

```diff
@@ -32,28 +41,110 @@ module Basics::Sandwich {
     /// Not enough funds to pay for the good in question
     const EINSUFFICIENT_FUNDS: u64 = 0;
 
+    /// On module init, create a grocery
+    fun init(ctx: &mut TxContext) {
+        Transfer::share_object(Grocery { 
+            id: TxContext::new_id(ctx),
+            profits: Coin::zero<SUI>(ctx)
+        });
+
+        Transfer::transfer(GroceryOwnerCapability {
+            id: TxContext::new_id(ctx)
+        }, TxContext::sender(ctx));
+    }
+
     /// Exchange `c` for some ham
-    public fun buy_ham(c: Coin<SUI>, ctx: &mut TxContext): Ham {
+    public fun buy_ham(grocery: &mut Grocery, c: Coin<SUI>, ctx: &mut TxContext): Ham {
         assert!(Coin::value(&c) == HAM_PRICE, EINSUFFICIENT_FUNDS);
-        Transfer::transfer(c, GROCERY);
+        Coin::join(&mut grocery.profits, c);
         Ham { id: TxContext::new_id(ctx) }
     }
 
     /// Exchange `c` for some bread
-    public fun buy_bread(c: Coin<SUI>, ctx: &mut TxContext): Bread {
+    public fun buy_bread(grocery: &mut Grocery, c: Coin<SUI>, ctx: &mut TxContext): Bread {
         assert!(Coin::value(&c) == BREAD_PRICE, EINSUFFICIENT_FUNDS);
-        Transfer::transfer(c, GROCERY);
+        Coin::join(&mut grocery.profits, c);
         Bread { id: TxContext::new_id(ctx) }
     }
 
     /// Combine the `ham` and `bread` into a delicious sandwich
-    public fun make_sandwich(
-        ham: Ham, bread: Bread, ctx: &mut TxContext
-    ) {
+    public fun make_sandwich(ham: Ham, bread: Bread, ctx: &mut TxContext) {
         let Ham { id: ham_id } = ham;
         let Bread { id: bread_id } = bread;
         ID::delete(ham_id);
         ID::delete(bread_id);
         Transfer::transfer(Sandwich { id: TxContext::new_id(ctx) }, TxContext::sender(ctx))
     }
+
+    /// See the profits of a grocery
+    public fun profits(grocery: &Grocery): u64 {
+        Coin::value(&grocery.profits)
+    }
+
+    /// Owner of the grocery can collect profits by passing his capability
+    public fun collect_profits(cap: GroceryOwnerCapability, grocery: &mut Grocery, ctx: &mut TxContext) {
```

Comment:
Can `cap` be `&mut`?

Reply:
You are right it can be just `ref`. Only the owner can pass it as parameter

Reply:
Thanks! I will fix it to ref then. This transfer thingy felt redundant and too explicit.